src/lib/steps.ts

```diff
@@ -21,6 +21,16 @@ export type AddTextColumnStep = {
   new_column: string;
 };
 
+export type AddTotalRowsStep = {
+  name: 'totals';
+  // an array of couples [<columnName>, <totalRowsLabel>]
+  totals: string[][];
+  // an array of columns used for groupby logic
+  groups?: string[];
+  /** the list of columnns to aggregate, with related aggregation function to use */
```

Comment:
True, I pushed a first commit to rename the interface for clarity (I chose `Aggregation` because it's actually more than just a function): de0933eb9bc5c0bd17962d03ade009173ffebbc4
(I already changed `AggregationStep` to `AggregateStep` to match the name of the step and distinguish it  bit more from `Aggregation`)